conan/tools/gnu/autotools.py

```diff
@@ -15,17 +17,39 @@ def __init__(self, conanfile, namespace=None):
                                                      namespace=namespace)
         self._configure_args = toolchain_file_content.get("configure_args")
         self._make_args = toolchain_file_content.get("make_args")
+        self.default_configure_install_args = True
+        self.build_script_folder = None
```

Comment:
This usage of ``build_script_folder`` as member is weird, it is passed as argument in the method, and only used inside that single method, it seems the member is unnecessary.

Reply:
Yes, it's weird, I did it because you need it for the configure inside the install and I wanted to avoid passing that in the constructor or in the install method, probably better to pass it in the constructor...